netkan/netkan/common.py

```diff
@@ -52,3 +70,89 @@ def download_stream_to_file(download_url: str, dest_file: IO[bytes]) -> None:
                       timeout=60) as req:
         for chunk in req.iter_content(chunk_size=8192):
             dest_file.write(chunk)
+
+
+class BaseMessageHandler:
+    STRATEGY_OPTION = 'ours'
+    game: Game
+
+    def __init__(self, game: Game) -> None:
+        self.game = game
+
+    # Apparently gitpython can be leaky on long running processes
+    # we can ensure we call close on it and run our handler inside
+    # a context manager
+    def __enter__(self) -> 'BaseMessageHandler':
+        if not self.repo.is_active_branch('master'):
+            self.repo.checkout_branch('master')
+        self.repo.pull_remote_branch(
+            'master', strategy_option=self.STRATEGY_OPTION)
```

Comment:
Can we do better than hard-coding `master` in new code now that we know we need to change it?

Reply:
Yes. I want to refactor our git usage in an upcoming PR.